### PR TITLE
Alerting: Forward user headers through loki/cortex proxied APIs

### PR DESCRIPTION
**Pull request by alexweav**
**What is this feature?**

In certain on-prem setups users might want to forward auth context to the loki/mimir/cortex behind grafana. Grafana does not forward request headers through the proxy API, making this impossible.

**Why do we need this feature?**

Enables certain on-prem setups with custom auth in front of an external ruler.

**Which issue(s) does this PR fix?**:

n/a

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/ngalert/api/util.go [hunk 1]
```diff
@@ -111,6 +111,11 @@ func (p *AlertingProxy) withReq(
 	if err != nil {
 		return ErrResp(http.StatusBadRequest, err, "")
 	}
+	for h, vs := range ctx.Req.Header {
+		for _, v := range vs {
+			req.Header.Add(h, v)
+		}
+	}
 	for h, v := range headers {
```
Comment (alexweav) [inline pkg/services/ngalert/api/util.go:119]: These headers are "extra" headers injected by grafana, not user headers. They are nil almost all the time, and rarely contain a Content-Type header

**Comment by diegommm** [comment]
Hi! Greetings from the Release Guild. Just letting you know I'm updating your PR's milestone to [11.2.x](https://github.com/grafana/grafana/milestone/529). If you need to see your changes in a previous version, please consider adding a backport label.

Thank you!

File: pkg/services/ngalert/api/util.go (hunk 1 above)
Comment (rwwiv) [reply pkg/services/ngalert/api/util.go:119]: Worth it to mention this in a comment, WDYT?

**Review by rwwiv — APPROVED** [review]
Quick nit, but LGTM

**Comment by yuri-tceretian** [comment]
I think this should fix https://github.com/grafana/grafana/issues/87386, right? What's remaining in this PR?

**Pull request closed without merging**